tests/test_train.py

```diff
@@ -109,3 +112,29 @@ def test_trainvarchanged(self):
         for b, a in zip(before, after):
             # make sure something changed
             assert (b.numpy() != a.numpy()).any()
+
+
+class TestMain(object):
```

Comment:
Shouldn't it be called `TestTrain`?

Reply:
arguable. I changed the name to TestTrainMain because we are testing the main function